### ISSUE TITLE
NodeJS ES Modules support

### ISSUE THREAD

**Issue by yanickrochon** [Urigo/graphql-scalars#681]
When upgrading to Node.js v14.5, I get into issues with this module because it does not recognize the exported module correctly. I have tried this solution with a small test project, and it seems to work :

## package.json

```
   "main": "dist/index.cjs.js",
   "type": "commonjs",
   "exports": {
      ".": [
         {
            "import": "dist/index.esm.js",
            "require": "dist/index.cjs.js",
            "default": "dist/index.cjs.js"
         },
         "dist/index.cjs.js"
      ]
   }
```

**Comment by yanickrochon** [comment]
Actually, I think having `"type": "module"` would lead to a better outcome. The `import`s need to have the file extensions, though.

**Comment by ardatan** [comment]
I think you are talking about ESM support but even GraphQLJS package doesn't support ESM very well so I don't think having this supported in this packages makes sense at least for now

**Comment by yanickrochon** [comment]
This is why I started by suggesting the CommonJS standard approach, then commented about actually taking the leap of adopting the latest adopted proposals in terms of Node modules. I have started to upgrade a few of my projects, only offering backward compatibility through Babel, and I think it's about time the community push hard on this side. I hope this issue can remain, and perhaps be linked to a roadmap.
